lambdas/process-journey-event/src/main/java/uk/gov/di/ipv/core/processjourneyevent/ProcessJourneyEventHandler.java

```diff
@@ -123,6 +127,14 @@ public Map<String, Object> handleRequest(Map<String, String> input, Context cont
                 sendMitigationStartAuditEvent(ipvSessionId, ipAddress, clientOAuthSessionItem);
             }
 
+            // If we have a new journey response, follow it
+            while (stepResponse instanceof JourneyStepResponse journeyResponse) {
+                ipvSessionItem.setJourneyType(journeyResponse.getJourneyType());
+                ipvSessionItem.setUserState(journeyResponse.getInitialState());
+                ipvSessionService.updateIpvSession(ipvSessionItem);
+                stepResponse = executeJourneyEvent(NEXT_EVENT, ipvSessionItem);
```

Comment:
We could also return this as an event and get the step function to re-run the lambda, it just feels needlessly inefficient.

Reply:
As discussed, there are some benefits to separating the two - e.g. ability to use feature flags etc.

The hardcoded `next` event is a bit unsatisfactory, but not much worse than what we have currently.